sokol_gfx.h

```diff
@@ -6435,6 +6436,13 @@ _SOKOL_PRIVATE void _sg_gl_apply_pipeline(_sg_pipeline_t* pip) {
     }
 }
 
+_SOKOL_PRIVATE void _sg_gl_shader_texture_slot(_sg_shader_t* shd, GLint loc, int slot) {
+    if(shd->gl.texture_slots[loc] != slot) {
```

Comment:
This array access will be a problem I think. The index is the GL uniform location, which isn't related to SG_MAX_SHADERSTAGE_IMAGES and might overflow.

I think the texture_slots array should go into _sg_gl_shader_stage_t struct, the array size remains at SG_MAX_SHADERSTAGE_IMAGES, but one entry needs to have two items:

- gl_loc: the GL uniform location (== loc function arg)
- gl_tex_slot: the GL texture slot (== slot argument)

(this should probably become a new struct _sg_gl_shader_stage_texture_slot).

You'd need two new parameters to the _sg_gl_shader_texture_slot() function:

- int stage: the shader stage index
- int img_index: the shader image index (or bind slot)

Both are available in the nested loop that calls the _sg_gl_shader_texture_slot() function.